docs/guides/real-time/chat/javascript.md

```diff
@@ -75,74 +104,64 @@ document.querySelector('#new').addEventListener('submit', function (event) {
 });
 ```
 
-## Establish WebSocket Connection
-
-Within the `#login` form submit event handler, extract the `email` and `password` values from the form:
+Within the `#login` form submit event handler, get access to the email and password values
 
 ```js
 const email = event.target.elements.email.value;
 const password = event.target.elements.password.value;
 ```
 
-Create a new WebSocket, which will immediately attempt connection:
+Now, call the login method on the client, passing the email and password
 
 ```js
-connection = new WebSocket(url);
-```
+document.querySelector('#login').addEventListener('submit', function (event) {
+	event.preventDefault();
 
-On connection, you must [send an authentication message before the timeout](/guides/real-time/authentication). Add an
-event handler for the connection's `open` event:
+  const email = event.target.elements.email.value;
+  const password = event.target.elements.password.value;
 
-```js
-connection.addEventListener('open', function () {
-	connection.send(
-		JSON.stringify({
-			type: 'auth',
-			email,
-			password,
-		})
-	);
+  client.login(email, password); // [!code ++]
 });
 ```
 
-## Subscribe To Messages
-
-In a WebSocket connection, all data sent from the server will trigger the connection’s `message` event. Underneath the
-`open` event handler, add the following:
+Once the client is authenticated, immediately create a WebSocket connection:
 
 ```js
-connection.addEventListener('message', function (message) {
-	receiveMessage(message);
-});
+client.connect();
 ```
 
-At the bottom of your `<script>`, create the `receiveMessage` function:
+## Subscribe To Messages
+
+As soon as you have successfully authenticated, a message will be sent. When this happens, subscribe to updates on the
+`Messages` collection.
 
 ```js
-function receiveMessage(message) {
-	const data = JSON.parse(message.data);
-}
+client.onWebSocket('message', function (message) {
+  if (data.type == 'auth' && data.status == 'ok') {
+	  subscribe('update')
```

Comment:
This isn't a complex enough guide to leave 'unresolved threads' like above - where you call a function and then write it. For this reason, I'd suggest flipping the order - define function, and then call it.

Reply:
I get what you mean, but it made sense to do it this way because of the sentence that preceded talking about what to do after authentication.